src/mpid/ch4/src/ch4_coll_select.h

```diff
@@ -20,71 +20,68 @@
 #endif
 
 
-MPL_STATIC_INLINE_PREFIX
-    MPIDI_coll_algo_container_t * MPIDI_CH4_Barrier_select(MPIR_Comm * comm,
-                                                           MPIR_Errflag_t * errflag)
+MPL_STATIC_INLINE_PREFIX const
+MPIDI_coll_algo_container_t *MPIDI_CH4_Barrier_select(MPIR_Comm * comm, MPIR_Errflag_t * errflag)
 {
     if (comm->comm_kind == MPIR_COMM_KIND__INTERCOMM) {
-        return (MPIDI_coll_algo_container_t *) & MPIDI_CH4I_Barrier_inter_composition_alpha_cnt;
```

Comment:
I agree with the removal of these casts, but that's not what is described by this commit message. IMO, they are a different logical change and should be a separate commit.